src/storage/aggregator.cc

```diff
@@ -143,48 +156,136 @@ bool Aggregator::Update(const std::string& key, const std::string& row, const ui
     return true;
 }
 
-bool Aggregator::GetAggrBuffer(const std::string& key, AggrBuffer* buffer) {
+bool Aggregator::Init() {
+    std::unique_lock<std::mutex> lock(mu_);
+    if (GetStat() != AggrStat::kUnInit) {
+        PDLOG(INFO, "aggregator status is %u", GetStat());
+        return true;
+    }
+    lock.unlock();
+    if (!base_replicator_) {
+        return false;
+    }
+    status_.store(AggrStat::kRecovering, std::memory_order_relaxed);
+    auto log_parts = base_replicator_->GetLogPart();
+
+    if (aggr_table_->GetRecordCnt() == 0 && log_parts->IsEmpty()) {
+        status_.store(AggrStat::kInited, std::memory_order_relaxed);
+        return true;
+    }
+    auto it = aggr_table_->NewTraverseIterator(0);
+    it->SeekToFirst();
+    uint64_t recovery_offset = UINT64_MAX;
+    uint64_t aggr_latest_offset = 0;
+    while (it->Valid()) {
+        AggrBufferLocked tmp_buffer;
+        auto& buffer = tmp_buffer.buffer_;
+        auto val = it->GetValue();
+        int8_t* aggr_row_ptr = reinterpret_cast<int8_t*>(const_cast<char*>(val.data()));
+        bool ok = GetAggrBufferFromRowView(aggr_row_view_, aggr_row_ptr, &buffer);
+        if (!ok) {
+            PDLOG(ERROR, "GetAggrBufferFromRowView failed");
+            status_.store(AggrStat::kUnInit, std::memory_order_relaxed);
+            return false;
+        }
+        recovery_offset = std::min(recovery_offset, buffer.binlog_offset_);
+        aggr_latest_offset = std::max(aggr_latest_offset, buffer.binlog_offset_);
+        int64_t latest_ts = buffer.ts_end_ + 1;
+        uint64_t latest_binlog = buffer.binlog_offset_ + 1;
+        buffer.clear();
+        buffer.ts_begin_ = latest_ts;
+        buffer.binlog_offset_ = latest_binlog;
+        if (window_type_ == WindowType::kRowsRange) {
+            buffer.ts_end_ = latest_ts + window_size_ - 1;
+        }
+        aggr_buffer_map_.emplace(it->GetPK(), std::move(tmp_buffer));
```

Comment:
`mutex` in AggrBufferLocked cannot to copy or move. refer [here](https://en.cppreference.com/w/cpp/thread/mutex)